src/components/page/elements/MultipleChoiceQuestionElement.js

```diff
@@ -0,0 +1,255 @@
+          <div onClick={this.handleNextClicked} className={'button has-text-left has-background-light is-fullwidth'}>
+            Next Question
+          </div>
+        )
+      }
+    } else {
+      textContent = 'Wrong.'
+      buttonContent = (
+        <div onClick={this.handleBackClicked} className={'button has-text-left has-background-light is-fullwidth'}>
+          Try again.
+        </div>
+      )
+    }
+    return (
+      <>
+        <div style={{ padding: '1rem 1.5rem', paddingBottom: '50px' }}>{textContent}</div>
+        <div>{buttonContent}</div>
+      </>
+    )
+  }
+
+  getHint() {
+    if (this.props.content.hints !== '' && this.state.showHints) {
+      return (
+        <div className={'has-background-warning'}>
+          <strong>Hint: </strong>
+          <ContentArray content={this.props.content.hints} />
+        </div>
+      )
+    }
+    return null
+  }
+
+  handleBackClicked() {
+    this.setState({ submitted: false })
+  }
+
+  handleNextClicked() {
+    console.log('next clicked')
+  }
+
+  handleQuestionSubmit() {
+    let answeredCorrectly = this.checkAnswers()
+    this.setState({ submitted: true, answeredCorrectly, showHints: !answeredCorrectly })
+    console.log('Your answer is ', answeredCorrectly)
+  }
+
+  handleInputChange(event) {
+    const target = event.target
+    const value = target.type === 'checkbox' ? target.checked : target.value
+    const name = target.name
+    if (this.props.content.multiplechoice) {
+      let newSelectedState = this.state.selected
+      if (value && newSelectedState.filter(itm => itm === name).length === 0) {
+        newSelectedState.push(name)
+      }
+      if (!value && newSelectedState.filter(itm => itm === name).length >= 1) {
+        newSelectedState = newSelectedState.filter(itm => itm !== name)
+      }
+      this.setState({ selected: newSelectedState })
+    } else {
+      this.setState({ selected: [value] })
+    }
+  }
+
+  checkAnswers() {
+    let answeredCorrectly = false
+    const correctAnswers = this.getCorrectAnswers()
+    if (!this.props.content.multiplechoice) {
+      answeredCorrectly = correctAnswers.length === 1 && correctAnswers[0] === this.state.selected[0]
+    } else {
+      if (correctAnswers.length === this.state.selected.length) {
+        answeredCorrectly = correctAnswers.reduce((acc, value) => {
+          return acc || this.state.selected.filter(itm => itm === value).length >= 1
+        }, false)
+      }
+    }
+    return answeredCorrectly
+  }
+
+  getCorrectAnswers() {
+    const correctAnswers = []
+    this.props.content.answers.forEach(answer => {
+      if (answer.content[0].value) {
+        if (this.props.content.multiplechoice) {
+          correctAnswers.push(
+            shajs('sha256')
+              .update(JSON.stringify(answer.content[0].answer))
+              .digest('hex'),
+          )
+        } else {
+          if (correctAnswers.length === 0) {
+            const answerHash = shajs('sha256')
```

Comment:
`shajs('sha256').update(...).digest('hex')` should be a function.

```javascript
x => shajs('sha256').update(x).digest('hex')
```

Reply:
It also looks like the same code is copy pasted. Isn't possible to create one condition on line `174` to avoid duplication?

Reply:
fixed